packages/polling-controller/src/PollingController.ts

```diff
@@ -0,0 +1,121 @@
+import { BaseControllerV2 } from '@metamask/base-controller';
+import type { RestrictedControllerMessenger } from '@metamask/base-controller';
+import type { NetworkClientId } from '@metamask/network-controller';
+import type { Json } from '@metamask/utils';
+import { v1 as random } from 'uuid';
+
+export type PollingCompleteType<N extends string> = {
+  type: `${N}:pollingComplete`;
+  payload: [string];
+};
+
+/**
+ * PollingController is an abstract class that implements the polling
+ * functionality for a controller. It is meant to be extended by a controller
+ * that needs to poll for data by networkClientId.
+ *
+ */
+export default abstract class ControllerPolling<
+  Name extends string,
+  State extends Record<string, Json>,
+  messenger extends RestrictedControllerMessenger<
+    Name,
+    any,
+    PollingCompleteType<Name> | any,
+    string,
+    string
+  >,
+> extends BaseControllerV2<Name, State, messenger> {
+  readonly #intervalLength = 1000;
```

Comment:
Should this be hardcoded? Or rather something the child class implements?

Reply:
I think we may want to make this a required param in the constructor. Frequency of polling is definitely dependent on the context. I'm not certain there is a good default value

Reply:
Good point. It is important that we preserve polling intervals (or at least don't change them without good reason) as we roll this out